src/data/matrix/basic.lean

```diff
@@ -353,6 +353,18 @@ lemma map_mul {L : matrix m n α} {M : matrix n o α}
 (L ⬝ M).map f = L.map f ⬝ M.map f :=
 by { ext, simp [mul_apply, ring_hom.map_sum], }
 
+/-- A version of `one_map` where `f` is a ring hom. -/
+@[simp] lemma ring_hom_map_one [decidable_eq n]
+  {β : Type w} [semiring β] (f : α →+* β) :
+  (1 : matrix n n α).map f = 1 :=
+one_map f.map_zero f.map_one
+
+/-- A version of `map_zero` where `f` is a ring hom. -/
+@[simp] lemma ring_hom_map_zero
+  {β : Type w} [semiring β] (f : α →+* β) :
+  (0 : matrix n n α).map f = 0 :=
+map_zero f.map_zero
```

Comment:
Worth adding the other homs too?
```suggestion
/-- A version of `map_zero` where `f` is a `zero_hom`. -/
@[simp] lemma zero_hom_map_zero
  {β : Type w} [semiring β] (f : zero_hom α β) :
  (0 : matrix n n α).map f = 0 :=
map_zero f.map_zero

/-- A version of `map_zero` where `f` is a `add_monoid_hom`. -/
@[simp] lemma add_monoid_hom_map_zero
  {β : Type w} [semiring β] (f : α →+ β) :
  (0 : matrix n n α).map f = 0 :=
map_zero f.map_zero

/-- A version of `map_zero` where `f` is a `add_equiv`. -/
@[simp] lemma add_equiv_map_zero
  {β : Type w} [semiring β] (f : α ≃+ β) :
  (0 : matrix n n α).map f = 0 :=
map_zero f.map_zero

/-- A version of `map_zero` where `f` is a `ring_hom`. -/
@[simp] lemma ring_hom_map_zero
  {β : Type w} [semiring β] (f : α →+* β) :
  (0 : matrix n n α).map f = 0 :=
map_zero f.map_zero

/-- A version of `map_zero` where `f` is a `ring_equiv`. -/
@[simp] lemma ring_equiv_map_zero
  {β : Type w} [semiring β] (f : α ≃+* β) :
  (0 : matrix n n α).map f = 0 :=
map_zero f.map_zero

/-- A version of `map_zero` where `f` is a `alg_hom`. -/
@[simp] lemma alg_hom_map_zero
  {β : Type w} [semiring β] [semimodule R α] [semimodule R β] (f : α →ₐ[R] β) :
  (0 : matrix n n α).map f = 0 :=
map_zero f.map_zero

/-- A version of `map_zero` where `f` is a `alg_equiv`. -/
@[simp] lemma alg_equiv_map_zero
  {β : Type w} [semiring β] [semimodule R α] [semimodule R β] (f : α ≃ₐ[R] β) :
  (0 : matrix n n α).map f = 0 :=
map_zero f.map_zero
```
(might be missing some type classes for the `alg_` ones).

Reply:
Is this a case where you could use the `is_add_monoid_hom` typeclass to automatically cover all of the relevant hom types (except `zero_hom` which no one really uses anyway)?

Reply:
The typeclass sounds like a good idea, adding a `map_zero` for each `foo_hom` doesn't sound scalable, and the typeclass doesn't cause `simp` issues here. There is an `instance (f : R →+* S) : is_add_monoid_hom f` (in `deprecated`) but apart from that the constructions are set up only from unbundled to bundled. I'll [revive this thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Typeclasses.20for.20morphisms).